scripts/FASUpdate.py

```diff
@@ -0,0 +1,215 @@
+
+import time
+import requests
+import pathlib
+import sys
+import os
+import json
+import argparse
+from os import listdir
+from os.path import exists, isfile, join
+from datetime import datetime
+
+RECIPES_DIR = "recipes"
+
+def getFilepath():
+    filepath = os.path.abspath(os.path.dirname(__file__)) + "/" + RECIPES_DIR
+    return filepath
+
+def getAvailableFiles(filepath):
+    print("Available files in " + filepath + ":")
+    onlyfiles = [f for f in listdir(filepath) if isfile(join(filepath, f))]
+    for f in listdir(filepath):
+      if isfile(join(filepath, f)):
+          if pathlib.Path(f).suffix == ".json":
+              print(f)
+
+def doAction(token, data, FAS_URL):
+    headers = {
+        'Content-Type': "application/json",
+        'cache-control': "no-cache",
+        'Authorization': f'Bearer {token}'
+    }
+    url = FAS_URL + "/actions"
+    payload = json.dumps(data)
+    print("JSON payload to FAS action command:\n" + payload)
+    ret = requests.request("POST", url, data=payload, headers=headers, verify=True)
+    if ret.status_code != 202:
+        print("ERROR: Return Code: " + str(ret.status_code))
+        print(ret.json())
+        return None
+    if "actionID" in ret.json():
+        actionID = ret.json()["actionID"]
+    else:
+        actionID = None
+    print("Action ID: " + actionID)
+    return actionID
+
+def doWatchAction(token, FAS_URL, actionID, watchtime):
+    headers = {
+        'cache-control': "no-cache",
+        'Authorization': f'Bearer {token}'
+    }
+    state = "none"
+    while state != "completed":
+        time.sleep(watchtime)
+        url = FAS_URL + "/actions/" + actionID + "/status"
+        ret = requests.request("GET", url, headers=headers, verify=True)
+        retjson = ret.json()
+        state = retjson["state"]
+        if state == "completed":
+            print("--------------------------- " + state + " ----------------------")
+        print("State: " + state + " Date: " + datetime.now().strftime("%m/%d/%Y %H:%M:%S"))
+        for oc in retjson["operationCounts"]:
+            count = retjson["operationCounts"][oc]
+            if count > 0:
+                print("> " + oc + ": " + str(count))
+        if "errors" in retjson.keys():
+            errors = retjson["errors"]
+            if len(errors) > 0:
+                print("ERRORS :")
+                print(errors)
+        if state == "completed":
+            print("--------------------------- COMPLETED ----------------------")
+
+# https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+def main():
+    token = os.environ.get('TOKEN')
+    if token is None or token == "":

```

Comment:
Lets take this out.  They can search the manual for what to do if the token isnt set.  Just print line 111